livekit-agents/livekit/agents/voice/io.py

```
@@ -50,13 +50,15 @@ class PlaybackFinishedEvent:
 
 
 class AudioSink(ABC, rtc.EventEmitter[Literal["playback_finished"]]):
-    def __init__(self, *, sample_rate: int | None = None) -> None:
+    def __init__(self, *, sample_rate: int | None = None, num_channels: int | None = None) -> None:
```

We can't add num_channels yet, we don't have audio remixing

---

Feel free to ignore the channels for now

---

Do all TTS always output with one channel? otherwise if user misconfigured the num_channel of TTS and RoomIO.AudioSource it will raise the error.

Maybe that's too much, just let it fail seems ok?

---

updated

---

Generally, all TTS outputs one channel (at least by default).
Let's fail for now it's easier

---

Yeah, removed for now. The `AudioSource` will fail if the number of channel mismatch.